analytic_tag_dimension_enhanced/models/account_analytic_dimension.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class AccountAnalyticDimension(models.Model):
+    _inherit = "account.analytic.dimension"
+
+    ref_model_id = fields.Many2one(
+        comodel_name="ir.model",
+        string="Ref Model",
+        help="Select model if you want to use it to create analytic tags, "
+        "each tag will have reference to the data record in that model.\n"
+        "For example, if you select Department (hr.department) then click "
+        "Create Tags button, tags will be created from each department "
+        " and also has resource_ref to the department record",
+    )
+    filtered_field_ids = fields.Many2many(
+        comodel_name="ir.model.fields",
+        string="Filtered by fields",
+        domain="[('model_id', '=', ref_model_id), ('ttype', '=', 'many2one')]",
+        help="Filtered listing tags by fields of this model, based on value "
+        "of selected analytic tags in working document",
+    )
+    required = fields.Boolean(
```

Comment:
Is it possible to change the field name to is_required ?

Reply:
I would object if there is no strict rule about this (or let me know if it is). It will need to backport / migration.

Reply:
THere is no strict rule @kittiu Its just that 'required' seems used in other places.